public/app/core/components/Login/ChangePassword.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { PureComponent, SyntheticEvent, ChangeEvent } from 'react';
+import { Tooltip } from '@grafana/ui';
+import appEvents from 'app/core/app_events';
+
+interface Props {
+  onSubmit: (pw: string) => void;
+  onSkip: Function;
+  className?: string;
+  focus?: boolean;
+}
+
+interface State {
+  newPassword: string;
+  confirmNew: string;
+  valid: boolean;
+}
+
+export class ChangePassword extends PureComponent<Props, State> {
+  private userInput: HTMLInputElement;
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      newPassword: '',
+      confirmNew: '',
+      valid: false,
+    };
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (!prevProps.focus && this.props.focus) {
+      this.focus();
+    }
+  }
+
+  focus() {
+    this.userInput.focus();
+  }
+
+  onSubmit = (e: SyntheticEvent) => {
+    e.preventDefault();
+
+    const { newPassword, valid } = this.state;
+    if (valid) {
+      this.props.onSubmit(newPassword);
+    } else {
+      appEvents.emit('alert-warning', ['New passwords do not match', '']);
+    }
+  };
+
+  onChange = (e: ChangeEvent<HTMLInputElement>) => {
+    // @ts-ignore
+    const { value } = this.state;
+    // @ts-ignore
+    this.setState({
```

Comment:
This ignore is because State has no index signature. I suggest you create two separate methods (onNewPasswordChange, onConfirmPasswordChange) that will call onChange with a partial state update. And then you won't have to rely on `e.target.name`. Relying on event.target.name is flaky as any typo in the name of the input will not be picked up by Typescript and can lead to silent errors.